### PR TITLE
Spellbook improvements

### PR DESCRIPTION
**Pull request by petchema**
Implements Up and Down buttons to rearrange spells, Sort button switching between alphabetic and increasing magicka cost orders, and spell renaming.
Forums: https://forums.dfworkshop.net/viewtopic.php?f=27&t=1168&start=230#p18576

They're still issues with scrolling not following selection when the list of spells grows longer than one screen, so this can be considered WIP (I'm not sure how to best implement that)

### CODE REVIEW

**Comment by petchema** [comment]
Added modification of spell icon
https://forums.dfworkshop.net/viewtopic.php?f=24&t=1360

**Comment by ajrb** [comment]
The changes you made to listbox, can you describe how they change the behaviour please?

This component is used in a lot of places, so we need to be sure the change is valid for these before it goes in.

File: Assets/Scripts/Game/UserInterface/ListBox.cs [hunk 1]
```diff
@@ -679,9 +679,18 @@ public void SelectNone()
 
         public void ScrollToSelected()
         {
-            scrollIndex = selectedIndex;
+            if (selectedIndex < 0)
```
[the remark this answers is not in the document]
Comment (petchema) [reply Assets/Scripts/Game/UserInterface/ListBox.cs]: Ok, maybe I need to explain what this change does exactly;
- you're free to scroll around, it won't affect selection;
- if something visible is selected, it usually won't scroll; The exception is, if at least 4 items are visible and you select the very first or very last item, the list will scroll to reveal the next item up or down;
- if something not visible is selected, the list will be scrolled up or down as little as possible so the item becomes visible, with a similar exception as described above.
It retrospect it's probably safer to make a new method for this behavior, and test in each case which behavior works best.

**Comment by petchema** [comment]
> Can you give me detail what this is? I'm not familiar with any issue with scrolling in listbox control.

Clicking on the up or down arrows of the spells list scrollbar selects the next or previous spell; But the list is never scrolled, so the selection can go off screen.

Also, RefreshSpellsList() calls ListBox.ClearItems() that resets the scrolling (and the selected item) to the top of the list, so in several place where the list needs to be refreshed (say, if you rename a spell), you loose the list current position.

**Comment by Interkarma** [comment]
>Clicking on the up or down arrows of the spells list scrollbar selects the next or previous spell; But the list is never scrolled, so the selection can go off screen.

Ah, I see the problem here, cheers. There's already provision for this in listbox control. All that needs to be changed is for up/down arrow clicks in spellbook to call listbox.SelectNext() and listbox.SelectPrevious() instead of just changing selection index. Basically spellbook isn't following other controls.

I'll send a quick change for that now.

**Comment by Interkarma** [comment]
> Also, RefreshSpellsList() calls ListBox.ClearItems() that resets the scrolling (and the selected item) to the top of the list, so in several place where the list needs to be refreshed (say, if you rename a spell), you loose the list current position.

I can definitely see a case for this be improved though, yeah. :)

**Comment by Interkarma** [comment]
This is very good, thank you! :)

**Pull request merged**